rocketpy/mathutils/function.py

```diff
@@ -1012,6 +1012,97 @@ def to_frequency_domain(self, lower, upper, sampling_frequency, remove_dc=True):
             extrapolation="zero",
         )
 
+    def short_time_fft(
+        self, lower, upper, sampling_frequency, window_size, step_size, remove_dc=True
+    ):
+        '''
+        Performs the Short-Time Fourier Transform (STFT) of the Function and
+        returns the result. The STFT is computed by applying the Fourier transform
+        to overlapping windows of the Function.
+
+        Parameters
+        ----------
+        lower : float
+            Lower bound of the time range.
+        upper : float
+            Upper bound of the time range.
+        sampling_frequency : float
+            Sampling frequency at which to perform the Fourier transform.
+        window_size : float
+            Size of the window for the STFT, in seconds.
+        step_size : float
+            Step size for the window, in seconds.
+        remove_dc : bool, optional
+            If True, the DC component is removed from each window before computing the Fourier transform.
+
+        Returns
+        -------
+        list of Function
+            A list of Functions, each representing the STFT of a window.
+
+        Examples
+        --------
+        >>> from rocketpy import Function
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+
+        >>> # Generate a signal with varying frequency
+        >>> T_x, N = 1 / 20, 1000  # 20 Hz sampling rate for 50 s signal
+        >>> t_x = np.arange(N) * T_x  # time indexes for signal
+        >>> f_i = 1 * np.arctan((t_x - t_x[N // 2]) / 2) + 5  # varying frequency
+        >>> signal = np.sin(2 * np.pi * np.cumsum(f_i) * T_x)  # the signal
+
+        >>> # Create a Function object
+        >>> time_domain = Function(np.array([t_x, signal]).T)
+        >>> # Perform the STFT
+        >>> stft_result = time_domain.short_time_fft(
+        ...     lower=0, upper=50, sampling_frequency=20,
+        ...     window_size=2, step_size=0.5
+        ... )
+
+        >>> # Plot the result
+        >>> fig, ax = plt.subplots(figsize=(10, 6))
+        >>> for window in stft_result:
+        ...     freq = window[:, 0]
+        ...     amplitude = window[:, 1]
+        ...     _=ax.plot(freq, amplitude)
+
+        >>> ax.set_xlabel('Frequency (Hz)')
+        Text(0.5, 0, 'Frequency (Hz)')
+        >>> ax.set_ylabel('Amplitude')
+        Text(0, 0.5, 'Amplitude')
+        >>> ax.set_title('Short-Time Fourier Transform (STFT) Result')
+        Text(0.5, 1.0, 'Short-Time Fourier Transform (STFT) Result')
+        >>> plt.show()
```

Comment:
It is hard to interpret this plot. The best approach to interpret the results is by plotting a spectrogram, as shown in SciPy's example or in the example notebook I provided in the previous review.